presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -454,7 +454,7 @@ default boolean supportsMissingColumnsOnInsert()
     /**
      * Begin materialized view query
      */
-    default ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle)
+    default ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, List<ConnectorTableHandle> sourceTableHandles)
```

Comment:
From what I can tell, the query for a materialized view can reference any connector. Passing arbitrary arbitrary table handles from other connectors is not useful (the connector can't do anything with them). It's also expected that a connector only gets their own table handles. Maybe we should filter this list to the connector and a boolean indicating that there are source tables for other connectors.

Reply:
Cross referencing connectors in a materialized view will need some work and API changes. The design is not yet completely mapped out, so I hesitate to make this change as it feels incomplete change that may need further changes down the line.
If we think this is a good change to have, I am open to it. FYI @martint for his input as well.